### ISSUE TITLE
testing channel auto updates fail for the past 3 versions

### ISSUE THREAD

**Issue by working-name** [MacDownApp/macdown#1029]
Hi!

First off, **great job** with MacDown, it has been a huge help in writing essays for applications, among others. Thank you again for releasing it for free and working on it.

Just a quick FYI:
For the past 3 version bumps on the testing/beta channel, the software fails to download the update.

It kinda becomes a nag and it makes me want to turn it off, but that's going to defeat the purpose of trying out new versions.

![updateproblem](https://user-images.githubusercontent.com/16505052/48314546-1a5f2480-e599-11e8-9fbc-ab7765bb5a8b.gif)

**Comment by manngo** [comment]
I have found the same.

What makes it more confusing is that I can’t find anyway of downloading the newer version and installing it manually.

**Comment by working-name** [comment]
You can find these links under History -> Testing on the site, but they’re not automatically posted, it’s manual. This new version is not yet up there. 

I will set up Charles Proxy and figure out what the actual requested url is when I get some time.

**Comment by FranklinYu** [comment]
This is possibly related to #900.

**Comment by working-name** [comment]
Well that's a bummer with d.pr being restrictive. As I mentioned above, if you go to History -> [Testing](https://macdown.uranusjr.com/history/testing/) you get the testing branch releases, but they're not automatically posted.

**Comment by FranklinYu** [comment]
> they're not automatically posted

I'm not sure I follow you.

**Comment by working-name** [comment]
> I'm not sure I follow you.

Someone has to manually post the latest version and associated changelog to the "Testing channel" page. It is not an automated process, which is why the latest version on the site is `Version 0.7.2d126` but the auto-updater says the latest version is actually `0.7.2d137`

The way I see it you can do 2 things: provide a link in the changelog in the auto update window, or automate the posting of newly compiled versions to the testing channel page.

**Comment by FranklinYu** [comment]
@working-name Good catch. @uranusjr messed up the order; you can find 0.7.2d137 at the *bottom* of "Testing channel" page.

It is indeed automatic, but the updater will comparing *all* the versions instead of simply looking for the one on top.

**Comment by manngo** [comment]
> @working-name Good catch. @uranusjr messed up the order; you can find 0.7.2d137 at the _bottom_ of "Testing channel" page.

Of course! Why didn’t I think of that … ?

Good one!
